alfasim_sdk/hook_specs.py

```diff
@@ -389,46 +1228,101 @@ def update_boundary_condition_of_mass_fraction_of_tracer(
+    the mass of the mixture. The ``vol_frac_bound`` is the volume fraction of fields at the boundary in which
+    the ``phi_presc`` is being calculated.
 
+    This `hook` allows the developer to update the boundary conditions of mass fraction that may depend on |alfasim|'s
+    internal variables that may change during the simulation. So, this update will be performed to each time step.
+
+    :param ctx: ALFAsim's plugins context
+    :param phi_presc: Prescribed mass fraction of tracer
+    :param tracer_index: Tracer ID
+    :param vol_frac_bound: Volume fraction of fields in the boundary
+    :param n_fields: Number of fields
     :returns: Return OK if successful or anything different if failed
-    """
 
+     Example of usage:
+
+    .. code-block:: c++
+        :linenos:
+        :emphasize-lines: 1
+
+        HOOK_UPDATE_BOUNDARY_CONDITION_OF_MASS_FRACTION_OF_TRACER(
+            void* ctx, void* phi_presc, int tracer_index)
+        {
+            // getting plugin internal data
+            int errcode = -1;
+            int thread_id = -1;
+            errcode = alfasim_sdk_api.get_thread_id(ctx, &thread_id);
+            if (errcode != OK) {
+                return errcode;
+            }
+            MyStruct* data = nullptr;
+            errcode = alfasim_sdk_api.get_plugin_data(
+                    ctx, (void**) &data, plugin_id, thread_id);
+            // Casting to double pointer
+            double* phi_presc_ptr = static_cast<double*>(phi_presc);
+            double* vol_frac_bound_ptr = static_cast<double*>(vol_frac_bound);
+            // Let suppose that this tracer is only in the gas field
+            if (tracer_index != data.my_user_defined_tracer_id) {
+                return OK;
+            } else {
+                // Update the boundary condition
+                // to the tracer mass fraction.
+                phi_presc_ptr =
+                        data.calc_bc_mass_fraction_of_tracer_in_gas_field(
+                            vol_frac_bound_ptr[data.continuous_gas_field_id],
+                            // List of params that can be
+                            // retrieved by get_simulation_array()
+                            );
+            }
+            return OK;
+        }
+
+    .. warning::
+        The plugin developer must NOT change ``vol_frac_bound`` variable, only the output variable ``phi_presc``.
 
-def initialize_mass_fraction_of_tracer(
-    ctx: "void*", phi_initial: "void*", tracer_index: "int"
-) -> "int":
     """
-    Internal tracer model Hook to initialize the mass fraction of tracer, given by `tracer_id`, in the entire
-    network. The output variable `phi_initial` is the initial mass fraction of the given tracer in respect
-    to the mass of the mixture.
 
-    :returns: Return OK if successful or anything different if failed
+
+def friction_factor(v1: "int", v2: "int") -> "int":
+    """
+    Docs for Friction Factor
```

Comment:
Missing docs here?

Reply:
I'm not sure if this hook is used

CC @ggrbill , @tarcisiofischer